### PR TITLE
added equb and subscription endpoints

### PR DESCRIPTION
**Pull request by ermiyas111**

### CODE REVIEW

File: lib/saving.js [hunk 1]
```diff
@@ -10,4 +10,9 @@ async function withdrawSaving() {
   return apiResponse;
 }
 
-module.exports = { createSaving, withdrawSaving };
+async function refund(request_ids) {
```
Comment (agebeyaw) [inline lib/saving.js]: try to rename this to claim, I will try to rename the API endpoint also to same. It is wrong pick and can be confusing

File: lib/saving.js
Comment (agebeyaw) [inline lib/saving.js:1]: `/saving/create` with action `withdraw` and amount is not implemented, please add it. The default action is `deposit`, and you can also explicitly add it along with the amount.

**Review by agebeyaw — CHANGES_REQUESTED** [review]

File: lib/saving.js
Comment (ermiyas111) [reply lib/saving.js:1]: Having an action parameter which is dynamic should handle both actions. Updating based on that

File: lib/saving.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const { apiRequest } = require('./apiRequest');
 
-async function createSaving(amount, deposit) {
-  const apiResponse = await apiRequest('POST', '/saving/create', '', { amount, deposit });
+async function createSaving(amount, action) {
```
Comment (agebeyaw) [inline lib/saving.js]: is it not possible to do this

```
async function createSaving(amount, action='deposit') {
const apiResponse = await apiRequest('POST', '/saving/create', '', { amount, action });
return apiResponse;
}
```

File: lib/transaction.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const { apiRequest } = require('./apiRequest');
 
-async function getTransactionListByUser() {
-  const apiResponse = await apiRequest('GET', '/transaction/find-by-user', '?p=1', null);
+async function getTransactionListByUser(param) {
```
Comment (agebeyaw) [inline lib/transaction.js]: same as above for param

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

File: lib/saving.js (hunk 2 above)
Comment (ermiyas111) [reply lib/saving.js]: Yeah, that should work better

File: lib/transaction.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const { apiRequest } = require('./apiRequest');
 
-async function getTransactionListByUser() {
-  const apiResponse = await apiRequest('GET', '/transaction/find-by-user', '?p=1', null);
+async function getTransactionListByUser(param='?p=1') {
+  const apiResponse = await apiRequest('GET', '/transaction/find-by-user', param, null);
```
Comment (agebeyaw) [inline lib/transaction.js]: btw, what about making the third argument as array and do create the query parameters dynamically from inside the `apiRequest` using the following?

```
new URLSearchParams(param).toString()
```

I think this way the developer does not have to worry about appending the query string with `?` among other complexities. what do you think?

Comment (ermiyas111) [reply lib/transaction.js]: Is it better to make it an array or key value pair object?

Comment (agebeyaw) [reply lib/transaction.js]: key value pair, correct

**Review by agebeyaw — APPROVED** [review]

**Pull request merged**